### DISCUSSION TITLE
Fails to generate code

### DISCUSSION THREAD

**Discussion by raskolnikovdd** [melih-unsal/DemoGPT#16] [Q&A]
I've repeatedly tried to generate an LLM-based demo but they all failed after the "refining code" message appears. I see that OpenAI is charging me for calls to the API.

**Comment by melih-unsal** [comment]
Hi @raskolnikovdd sorry to hear that. 
Currently, the model is under development but for the next release which will be published within 10 days, it will be better in terms of the comprehension and capability and also we will add max iteration count to limit the number of iterations to avoid overuse but currently, the max iteration is set to default value as 10 for test purpose.
If you share the demo ideas you tried, it would be awesome to work on them to accelerate the capability. 🙏🏻

**Reply by raskolnikovdd** [reply]
Cool, I want to create a demo that can guide a user through completing an occupational health and safety risk assessment. The user names the task, then they are prompted to select associated hazards, to assign risk ranking to each hazard, and finally to define controls.

This may be too complex, but a parred down version might just be something that helps the user identify hazards with a given task.
